### PR TITLE
add reset-submodule script

### PR DESCRIPTION
**Pull request by BusyJay**
So when switching branch/tag, we can cleanup workspace conveniently.

### CODE REVIEW

File: scripts/reset-submodule [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+git submodule update --init grpc-sys/grpc
+cd grpc-sys/grpc
+git submodule update --init third_party/boringssl
+git submodule update --init third_party/cares/cares
+git submodule update --init third_party/zlib
+cd third_party/zlib
```
Comment (siddontang) [inline scripts/reset-submodule]: why cd zlib here?

File: scripts/reset-submodule [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+git submodule update --init grpc-sys/grpc
+cd grpc-sys/grpc
+git submodule update --init third_party/boringssl
```
Comment (siddontang) [inline scripts/reset-submodule]: can we use `git submodule update --init --recursive` directly?

Comment (BusyJay) [reply scripts/reset-submodule]: Then many unused submodules get initialized, which can be slow and probably fail due to network censor.

File: scripts/reset-submodule (hunk 1 above)
Comment (BusyJay) [reply scripts/reset-submodule]: See madler/zlib#162

**Comment by siddontang** [comment]
does the script have the executive mode?

**Comment by BusyJay** [comment]
Yes.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by siddontang** [comment]
PTAL @overvenus

**Pull request merged**